File: Frontend/public/src/services/GradeService.js
import axios from "axios";

const GRADE_API_BASE_URL = 'http://localhost:8080/api/grades';

class GradeService {
    getAllGrades() {
        return axios.get(GRADE_API_BASE_URL);
    }

    getGradeById(id) {
        return axios.get(GRADE_API_BASE_URL + '/' + id);
    }

    createGrade(grade) {
        return axios.post(GRADE_API_BASE_URL, grade);
    }

    updateGrade(id, grade) {
        return axios.put(GRADE_API_BASE_URL + '/' + id, grade);
    }

    deleteGrade(id) {
        return axios.delete(GRADE_API_BASE_URL + '/' + id);
    }

    getGradesByStudentId(studentId) {
        return axios.get(GRADE_API_BASE_URL + '/student/' + studentId);
    }

    getGradesByCourseId(courseId) {
        return axios.get(GRADE_API_BASE_URL + '/course/' + courseId);
    }

    getStudentGPA(studentId) {
        return axios.get(GRADE_API_BASE_URL + '/gpa/' + studentId);
    }
}

export default new GradeService(); 

File: School-Management-System-2025/src/main/java/com/rehan/sms/services/GradeService.java
package com.rehan.sms.services;

import com.rehan.sms.dto.GradeDto;
import com.rehan.sms.exception.ResourceNotFoundException;
import java.util.List;

public interface GradeService {
    List<GradeDto> getAllGrades();

    GradeDto getGradeById(Long id) throws ResourceNotFoundException;

    GradeDto createGrade(GradeDto gradeDto);

    GradeDto updateGradeById(Long id, GradeDto gradeDto) throws ResourceNotFoundException;

    void deleteGradeById(Long id) throws ResourceNotFoundException;

    List<GradeDto> getGradesByStudentId(Long studentId);

    List<GradeDto> getGradesByCourseId(Long courseId);

    List<GradeDto> getGradesByStudentIdAndCourseId(Long studentId, Long courseId);
}